packages/transport/src/modules/relationships/local/PeerDeletionInfo.ts

```diff
@@ -20,6 +23,17 @@ export class PeerDeletionInfo extends Serializable implements IPeerDeletionInfo
     })
     public deletionStatus: PeerDeletionStatus;
 
+    @serialize()
+    @validate()
+    public deletionDate: CoreDate;
+
+    public static override preFrom(value: any): any {
+        if (!value.deletionDate) {
+            value.deletionDate = value.deletionStatus === PeerDeletionStatus.ToBeDeleted ? CoreDate.local().add({ days: 14 }) : CoreDate.local();
```

Comment:
two things

1. switch over `value.deletionStatus as PeerDeletionStatus` to make sure the switch is exhaustive if we add another status
2. in each block do `CoreDate#toString()` to be closer to having the thing in the DB.